pkg/service/service.go

```diff
@@ -160,13 +196,122 @@ func NewService(monitorNotify monitorNotify) *Service {
 		backendRefCount: counter.StringCounter{},
 		backendByHash:   map[string]*lb.Backend{},
 		monitorNotify:   monitorNotify,
+		envoyCache:      envoyCache,
 		healthServer:    localHealthServer,
 		lbmap:           lbmap.New(maglev, maglevTableSize),
+		l7lbSvcs:        map[ServiceName]*L7LBInfo{},
 	}
 	svc.lastUpdatedTs.Store(time.Now())
+
 	return svc
 }
 
+// RegisterL7LBService makes the given service to be locally forwarded to the
+// given proxy port, if non-zero, or for backend sync only, if zero.
```

Comment:
I think it would be more intuitive to make different calls that represent what's actually going on, instead of having the port being zero or not present different functionality (even if it's quite similar). As a reader, it would make it easier to read and understand if there was a function for registering local forwarding and another one for backend sync only.

Reply:
Added a new `RegisterL7LBServiceBackendSync()` that does not take the zero port as an argument.